tensorflow/core/kernels/reshape_op.h

```diff
@@ -135,6 +136,17 @@ class ReshapeOp : public OpKernel {
         shape->AddDim(size);
         *has_zero_dim = true;
       } else {
+        if (MultiplyWithoutOverflow(shape->num_elements(), size) < 0) {
+          string msg;
+          for (int ii = 0; ii < num_dims; ++ii) {
+            if (ii != 0) {
+              strings::StrAppend(&msg, ", ");
+            }
+            strings::StrAppend(&msg, Svec(ii));
+          }
+          return errors::InvalidArgument("Shape [", msg,
+                                         "] has too many elements");
+	}
```

Comment:
The indentation here looks wrong; might be a spaces (the rest of the file) vs. tabs issue?